site/src/components/Resources/AppLink/AppLink.tsx

```diff
@@ -93,75 +93,65 @@ export const AppLink: FC<AppLinkProps> = ({ app, workspace, agent }) => {
+        css={{
+          pointerEvents: canClick ? undefined : "none",
+          textDecoration: "none !important",
+        }}
+        onClick={
+          canClick
+            ? async (event) => {
+                event.preventDefault();
+                // This is an external URI like "vscode://", so
+                // it needs to be opened with the browser protocol handler.
+                if (app.external && !app.url.startsWith("http")) {
+                  // If the protocol is external the browser does not
+                  // redirect the user from the page.
 
-                    // This is a magic undocumented string that is replaced
-                    // with a brand-new session token from the backend.
-                    // This only exists for external URLs, and should only
-                    // be used internally, and is highly subject to break.
-                    const magicTokenString = "$SESSION_TOKEN";
-                    const hasMagicToken = href.indexOf(magicTokenString);
-                    let url = href;
-                    if (hasMagicToken !== -1) {
-                      setFetchingSessionToken(true);
-                      const key = await getApiKey();
-                      url = href.replaceAll(magicTokenString, key.key);
-                      setFetchingSessionToken(false);
-                    }
-                    window.location.href = url;
-                  } else {
-                    window.open(
-                      href,
-                      Language.appTitle(
-                        appDisplayName,
-                        generateRandomString(12),
-                      ),
-                      "width=900,height=600",
-                    );
+                  // This is a magic undocumented string that is replaced
+                  // with a brand-new session token from the backend.
+                  // This only exists for external URLs, and should only
+                  // be used internally, and is highly subject to break.
+                  const magicTokenString = "$SESSION_TOKEN";
+                  const hasMagicToken = href.indexOf(magicTokenString);
+                  let url = href;
+                  if (hasMagicToken !== -1) {
+                    setFetchingSessionToken(true);
+                    const key = await getApiKey();
+                    url = href.replaceAll(magicTokenString, key.key);
+                    setFetchingSessionToken(false);
                   }
+                  window.location.href = url;
+                } else {
+                  window.open(
+                    href,
+                    Language.appTitle(appDisplayName, generateRandomString(12)),
+                    "width=900,height=600",
+                  );
                 }
-              : undefined
+              }
+            : undefined
+        }
+      >
+        <span
+          css={
+            !isPrivateApp && {
+              marginRight: 8,
+            }
```

Comment:
Can we eliminate the need for this span by pulling this CSS into the enclosing `<Link` component, or is it needed for structure?

Reply:
I was able to remove it 👍